File: src/typescript/check/input-validation/processUserAge.ts
function processUserAge(age: any): void {
    const numericAge = Number(age);
    if (numericAge >= 18) {
        console.log("User is an adult.");
    } else {
        console.log("User is a minor.");
    }
}

processUserAge("twenty");
processUserAge(null);
\ No newline at end of file

File: truth_set/go/check/input-validation/processUserAge.go
package main

import (
    "fmt"
    "strconv"
)

func processUserAge(age interface{}) {
    var numericAge float64
    
    switch v := age.(type) {
    case string:
        n, _ := strconv.ParseFloat(v, 64)
        numericAge = n
    case nil:
        numericAge = 0
    default:
        numericAge = 0
    }

    if numericAge >= 18 {
        fmt.Println("User is an adult.")
    } else {
        fmt.Println("User is a minor.") 
    }
}

func main() {
    processUserAge("twenty")
    processUserAge(nil)
}
\ No newline at end of file